File: SDKs/amazonaws.com/globalaccelerator/2018-08-08/java/lib/src/main/java/openapisdk/models/shared/UpdateListenerRequest.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

public class UpdateListenerRequest {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("ClientAffinity")
    public ClientAffinityEnum clientAffinity;
    public UpdateListenerRequest withClientAffinity(ClientAffinityEnum clientAffinity) {
        this.clientAffinity = clientAffinity;
        return this;
    }
    @JsonProperty("ListenerArn")
    public String listenerArn;
    public UpdateListenerRequest withListenerArn(String listenerArn) {
        this.listenerArn = listenerArn;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("PortRanges")
    public PortRange[] portRanges;
    public UpdateListenerRequest withPortRanges(PortRange[] portRanges) {
        this.portRanges = portRanges;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("Protocol")
    public ProtocolEnum protocol;
    public UpdateListenerRequest withProtocol(ProtocolEnum protocol) {
        this.protocol = protocol;
        return this;
    }
}

File: SDKs/amazonaws.com/globalaccelerator/2018-08-08/python/src/sdk/models/shared/updatelistenerrequest.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import clientaffinity_enum as shared_clientaffinity_enum
from ..shared import portrange as shared_portrange
from ..shared import protocol_enum as shared_protocol_enum


@dataclass_json
@dataclasses.dataclass
class UpdateListenerRequest:
    listener_arn: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('ListenerArn') }})
    client_affinity: Optional[shared_clientaffinity_enum.ClientAffinityEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('ClientAffinity') }})
    port_ranges: Optional[list[shared_portrange.PortRange]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('PortRanges') }})
    protocol: Optional[shared_protocol_enum.ProtocolEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Protocol') }})
    
